### PR TITLE
Various haddock fixes

### PR DESCRIPTION
**Pull request by jhrcek**

### CODE REVIEW

File: servant/src/Servant/API/Raw.hs [hunk 1]
```diff
@@ -11,6 +11,7 @@ import           Data.Typeable
 -- a modified (stripped) 'pathInfo' if the 'Application' is being routed with 'Servant.API.Sub.:>'.
 --
 -- In addition to just letting you plug in your existing WAI 'Application's,
--- this can also be used with <https://hackage.haskell.org/package/servant-server/docs/Servant-Utils-StaticFiles.html#v:serveDirectory serveDirectory> to serve
--- static files stored in a particular directory on your filesystem
+-- this can also be used with functions from
+-- <https://hackage.haskell.org/package/servant-server-0.16.2/docs/Servant-Server-StaticFiles.html Servant.Server.StaticFiles>
```
Comment (jhrcek) [inline servant/src/Servant/API/Raw.hs]: Replace link to deprecated module (Servant.Utils.StaticFiles) with the link to newer module (Servant.Server.StaticFiles)

Comment (phadej) [inline servant/src/Servant/API/Raw.hs]: Note that old link didn't had a version number.

Comment (jhrcek) [reply servant/src/Servant/API/Raw.hs]: You're right. Fixed

**Comment by jhrcek** [comment]
The CI build failed just on one configuration because of timeout, which seems unrelated to my PRs changes.

**Comment by phadej** [comment]
Let's see if restart would help. (Other take just 10min, so 50min overtime is suspicious)

**Comment by jhrcek** [comment]
Green

**Pull request merged**